From 7285abe69359392f3ec728d0fa8184b28fb922fc Mon Sep 17 00:00:00 2001
From: Han Wang <416810799@qq.com>
Date: Fri, 11 Jan 2019 12:29:57 +0100
Subject: [PATCH 6/9] LineageParts: Trust: Fix encryption status for legacy
 devices

 * Or with no password set, legacy encryption devices would
   always display "not encrypted".

Change-Id: Iedc1e6206deb40031f1beb034f631a51861eb830
---
 .../lineageos/lineageparts/trust/TrustPreferences.java    | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/org/lineageos/lineageparts/trust/TrustPreferences.java b/src/org/lineageos/lineageparts/trust/TrustPreferences.java
index 6ea800a..63a492f 100644
--- a/src/org/lineageos/lineageparts/trust/TrustPreferences.java
+++ b/src/org/lineageos/lineageparts/trust/TrustPreferences.java
@@ -190,11 +190,11 @@ public class TrustPreferences extends SettingsPreferenceFragment {
             summary = R.string.trust_feature_encryption_value_enabled;
         } else if (level == TrustInterface.TRUST_FEATURE_LEVEL_POOR) {
             icon = R.drawable.ic_trust_encryption_poor;
-            summary = isLegacy ?
-                R.string.trust_feature_encryption_value_disabled :
-                R.string.trust_feature_encryption_value_nolock;
+            summary = R.string.trust_feature_encryption_value_nolock;
         } else {
-            icon = R.drawable.ic_trust_encryption_bad;
+            icon = isLegacy ?
+                R.drawable.ic_trust_encryption_poor :
+                R.drawable.ic_trust_encryption_bad;
             summary = R.string.trust_feature_encryption_value_disabled;
         }
         mEncryptionPref.setIcon(icon);
-- 
2.17.1

